xc/xc7/tests/iologic_packing/CMakeLists.txt

```diff
@@ -0,0 +1,205 @@
+  # IOB+OLOGIC for the given IOB type and ILOGIC/OLOGIC mode of operation
+  #
+  # Allowed IOB_TYPE values are:
+  #  - IBUF
+  #  - IBUFDS
+  #  - OBUF
+  #  - OBUFT
+  #  - OBUFDS
+  #  - OBUFTDS
+  #  - IOBUF
+  #  - IOBUFDS
+  #
+  # Allowed ILOGIC operation modes:
+  #  - NONE (or not given)
+  #  - ISERDES
+  #
+  # Allowed OLOGIC operation modes:
+  #  - NONE (or not given)
+  #  - OSERDES
+  #
+  # TODO: Add support for IDDR and ODDR
+  #
+  # The BOARD parameter specifies the pinout variant. Please refer to the
+  # generate.py script for the list of legal values.
+  #
+  # IMPORTANT: Please also note that designs generated by this function are
+  # not meant to run on an actual hardware!
+  #
+
+    # Parse arguments
+    set(OPTION_ARGS
+    )
+    set(ONE_VALUE_ARGS
+        BOARD
+        IOB_TYPE
+        USE_IDELAY
+        ILOGIC_MODE
+        OLOGIC_MODE
+    )
+    set(MULTI_VALUE_ARGS
+    )
+
+    cmake_parse_arguments(
+        XC7_IOLOGIC_DESIGN
+        "${OPTION_ARGS}"
+        "${ONE_VALUE_ARGS}"
+        "${MULTI_VALUE_ARGS}"
+        ${ARGN}
+    )
+
+    # Get args
+    set(BOARD       ${XC7_IOLOGIC_DESIGN_BOARD})
+    set(IOB_TYPE    ${XC7_IOLOGIC_DESIGN_IOB_TYPE})
+    set(USE_IDELAY  ${XC7_IOLOGIC_DESIGN_USE_IDELAY})
+    set(ILOGIC_MODE ${XC7_IOLOGIC_DESIGN_ILOGIC_MODE})
+    set(OLOGIC_MODE ${XC7_IOLOGIC_DESIGN_OLOGIC_MODE})
+
+    # Format the design name
+    set(DESIGN iologic_${IOB_TYPE})
+    if (${USE_IDELAY})
+        set(DESIGN ${DESIGN}_IDELAY)
+    endif()
+    if (ILOGIC_MODE)
+        set(DESIGN ${DESIGN}_${ILOGIC_MODE})
+    endif()
+    if (OLOGIC_MODE)
+        set(DESIGN ${DESIGN}_${OLOGIC_MODE})
+    endif()
+
+    # Generate the design
+    set(VERILOG_FILE ${DESIGN}.v)
+    set(PCF_FILE ${DESIGN}.pcf)
+    set(XDC_FILE ${DESIGN}.xdc)
+
+    add_file_target(FILE ${VERILOG_FILE} GENERATED)
+    add_file_target(FILE ${PCF_FILE} GENERATED)
+    add_file_target(FILE ${XDC_FILE} GENERATED)
+
+    set(IOLOGIC_OPTS "")
+    if (${USE_IDELAY})
+        set(IOLOGIC_OPTS ${IOLOGIC_OPTS} "--idelay")
+    endif()
+    if ("${ILOGIC_MODE}" STREQUAL "ISERDES")
+        set(IOLOGIC_OPTS ${IOLOGIC_OPTS} "--iserdes")
+    endif()
+    if ("${OLOGIC_MODE}" STREQUAL "OSERDES")
+        set(IOLOGIC_OPTS ${IOLOGIC_OPTS} "--oserdes")
+    endif()
+
+    add_custom_command(
+        OUTPUT ${VERILOG_FILE} ${PCF_FILE} ${XDC_FILE}
+        COMMAND ${PYTHON3} ${CMAKE_CURRENT_SOURCE_DIR}/generate.py
+            --board ${BOARD}
+            --iob ${IOB_TYPE}
+            -o ${DESIGN}
+            ${IOLOGIC_OPTS}
+        DEPENDS ${PYTHON3} ${PYTHON3_TARGET} generate.py
+        )
+
+    # Add target
+    add_fpga_target(
+        NAME ${DESIGN}
+        BOARD ${BOARD}
+        INPUT_IO_FILE ${PCF_FILE}
+        INPUT_XDC_FILE ${XDC_FILE}
+        SOURCES ${VERILOG_FILE}
+        EXPLICIT_ADD_FILE_TARGET
+        # Enable non-dedicated routes for clocks in fasm2bels until the problem
```

Comment:
The bug is in the verilog from generate.py.  Don't use logic inverters for CLKB signals.  The inverter on CLK <-> CLKB is fixed and non-configurable.

Reply:
Removed the flag and fixed `generate.py` so the same clock goes to CLK and CLKB.